support/agent/miner_agent.go

```diff
@@ -275,6 +317,9 @@ func (ma *MinerAgent) createFault(v SimState) ([]message, error) {
 	faultNumber, ma.liveSectors = PopRandom(ma.liveSectors, ma.rnd)
 	ma.faultySectors = append(ma.faultySectors, faultNumber)
 
+	// avoid trying to upgrade a faulty sector
+	ma.ccSectors = filterSlice(ma.ccSectors, map[uint64]bool{faultNumber: true})
```

Comment:
Should we add back on recovery?

Reply:
Getting a few more sectors to upgrade usually won't affect the sim that much. I did attempt this just to show that a recovered sector could be upgraded, but it ended up requiring substantial additional state management in the miner agent, so I backed out for now.